app/models/katello/sync_plan.rb

```diff
@@ -56,7 +56,7 @@ def save_with_logic!
 
     def update_attributes_with_logics!(params)
       transaction do
-        params["cron_expression"] = '' unless params["interval"].eql? CUSTOM_CRON
+        params["cron_expression"] = '' if (params.key?("interval") && !params["interval"].eql?(CUSTOM_CRON) && self.interval.eql?(CUSTOM_CRON))
```

Comment:
can you add a test to test this scenario?

Reply:
Added some tests and also removed self.save! from update method as discussed offline.